app/src/test/java/com/monstarlab/features/auth/domain/AuthRepositoryTest.kt

```diff
@@ -0,0 +1,79 @@
+package com.monstarlab.features.auth.domain
+
+import com.monstarlab.features.auth.AuthTokenMapper
+import com.monstarlab.features.auth.data.api.AuthApi
+import com.monstarlab.features.auth.data.api.dtos.TokenResponseDTO
+import com.monstarlab.features.auth.domain.models.AuthToken
+import io.mockk.*
+import junit.framework.TestCase.assertEquals
+import kotlinx.coroutines.runBlocking
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+
+/**
+ * Created by aslam on 08,March,2023
+ */
+internal class AuthRepositoryTest {
+
+    private lateinit var mockAuthApi: AuthApi
+    private lateinit var mockAuthTokenMapper: AuthTokenMapper
+    private lateinit var authRepository: AuthRepository
+
+    @Before
+    fun setUp() {
+        mockAuthApi = mockk()
+        mockAuthTokenMapper = mockk()
+        authRepository = AuthRepository(mockAuthApi, mockAuthTokenMapper)
+    }
+
+    @After
+    fun tearDown() {
+        clearAllMocks()
+    }
+
+    @Test
+    fun testLoginReturnsExpectedAuthToken() = runBlocking {
+        // Arrange
+        val email = "aslam@test.com"
+        val password = "password"
+        val expectedTokenResponseDTO = TokenResponseDTO("token")
+        val expectedAuthToken = AuthToken("token")
+        coEvery { mockAuthApi.postLogin(email, password) } returns expectedTokenResponseDTO.also {
+            every {
+                mockAuthTokenMapper.map(it)
+            } returns expectedAuthToken
+        }
+
+        // Act
+        val actualAuthToken = authRepository.login(email, password)
+
+        // Assert
+        assertEquals(expectedAuthToken, actualAuthToken)
+        coVerify { mockAuthApi.postLogin(email, password) }.also {
+            verify {
+                mockAuthTokenMapper.map(expectedTokenResponseDTO)
+            }
+        }
+    }
+
+    /**
+     * A more appropriate approach would be to modify the repository layer to handle exceptions
+    that may occur during login operations, and then write tests that verify that the repository
+    layer properly handles these exceptions. This way, the code is more resilient and can handle
+    unexpected situations that may occur during the login process.
+     * Test to ensure that a network error is thrown when logging in and
+    the repository layer does not throw one.
+     */
+    @Test
+    fun testLoginNetworkErrorThrowsException() = runBlocking {
+        // The repository should throw a Network Error when there is a problem with the network connection.
+    }
+
+    // Test to ensure that a server error is thrown when logging in and the repository layer does not throw one.
+    @Test
+    fun testLoginServerErrorThrowsException() = runBlocking {
+        // The repository should throw a Server Error when there is a problem with the server.
+    }
+
+}
```

Comment:
The repository technically  layer technically throws the exceptions :)
They will be thrown [here first](https://github.com/aslamhossin/android-template/blob/feature/auth_unit_test/app/src/main/java/com/monstarlab/core/network/errorhandling/ApiErrorInterceptor.kt) 
and since there is no try-catch here, the repository will throw it as well, then it will be either handled in the case or in a ViewModel

So tests like these will  actually be useful only if you are using a proper MockWebServer and enqueue real Responses with the `ApiErrorInterceptor` setup